autode/config.py

```diff
@@ -368,6 +374,15 @@ def __setattr__(self, key, value):
         if key == 'max_core':
             value = Allocation(value).to('MB')
 
+        if key == 'freq_scale_factor':
+
+            if value is not None:
+                if not (0. < value <= 1.):
+                    raise ValueError("Cannot set the frequency scale factor "
+                                     "outside of (0, 1]")
+
+                value = float(value)
```

Comment:
On line 390 you use a non-floated version of value, and is only set as a float if it passes the if statement on line 384. Could it be better to set it as a value first, then run through the if statements?

Reply:
not really, as in once case it's a distance (with units) and the other a coefficient